File: Easy/Python/1863. Sum of All Subset XOR Totals.py
class Solution(object):
    def subsetXORSumHelper(self, nums, i, j):
        if i == len(nums):
            return j
        return self.subsetXORSumHelper(nums, i + 1, j^nums[i]) + self.subsetXORSumHelper(nums, i + 1, j)

    def subsetXORSum(self, nums):
        return self.subsetXORSumHelper(nums, 0, 0)

File: Easy/JavaScript/1863. Sum of All Subset XOR Totals.js
var subsetXORSum = function(nums) {
    return subsetXORSumHelper(nums, 0, 0);
};

var subsetXORSumHelper = function(nums, i, j){
    if(i == nums.length)
        return j;
    return subsetXORSumHelper(nums, i + 1, j^nums[i]) + subsetXORSumHelper(nums, i + 1, j);
};
